src/schema/posts.ts

```diff
@@ -3493,6 +3516,47 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return getPostById(ctx, info, savedPost.id);
     },
+    votePoll: async (
+      _,
+      args: { optionId: string; postId: string },
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLPost> => {
+      const userPost = await ctx.con.getRepository(UserPost).findOne({
+        where: {
+          postId: args.postId,
+          userId: ctx.userId,
+        },
+        relations: ['post'],
+      });
+
+      let post: PollPost | undefined;
+      if (userPost) {
+        post = (await userPost.post) as PollPost;
+      } else {
+        post = await ctx.con.getRepository(PollPost).findOneByOrFail({
+          id: args.postId,
+        });
+      }
```

Comment:
why do you need double logic for this, would below work?

```suggestion
      const post: PollPost | null = await userPost.post;
      
      if (!post) {
          throw new NotFoundError()
      }
```

Reply:
@capJavert If the userPost does not exist, we need to still fetch the post

Reply:
why not make it other way around, fetch the post and userPost as relation, then its a single line?

Reply:
ignore me, ok 👍